tests/test_update_manager_file_cache.py

```
@@ -0,0 +1,449 @@
+    @pytest.mark.asyncio
+    async def test_record_count_exception_handling(self, manager, temp_dir):
+        """Test record counting exception handling (lines 606-608)."""
+        # Create a directory with invalid permissions or corrupted file
+        invalid_file = temp_dir / "products.csv"
+        invalid_file.write_text("invalid,csv\n")
+
+        # Mock open to raise exception
+        with pytest.MonkeyPatch.context() as mp:
+            mp.setattr(
+                "builtins.open",
+                lambda *args, **kwargs: (_ for _ in ()).throw(IOError("Permission denied")),
```

Ruff (0.14.1)

Unused lambda argument: `args`

(ARG005)